### DISCUSSION TITLE
How can I zoom in and zoom out on a timeline, so that slider slides through less time when zoomed in and vice versa?

### DISCUSSION THREAD

**Discussion by YousufSSyed** [mifi/lossless-cut#1638] [Q&A] [answered]
I found this [issue](https://github.com/mifi/lossless-cut/issues/113) that asks about it but doesn't go over how to zoom in and out of the timeline.

**Comment by mifi** [comment, marked as answer]
you can use your mouse wheel or the zoom button in the lower left

**Reply by YousufSSyed** [reply]
How would I use my Macbook trackpad?

**Reply by mifi** [reply]
i use the pinch gesture

**Reply by YousufSSyed** [reply]
I tried selecting the timeline and pinching like this, and nothing happened:
![HT4721-Pinch-003-en](https://github.com/mifi/lossless-cut/assets/42646062/3b92dff1-5361-4022-8984-5c3b036f88f6)

How would I pinch, and what would it look like if it worked?

**Reply by mifi** [reply]
so it doesn't zoom in when you pinch like that after opening a file? what happens if you click the zoom button in the lower left?

**Reply by YousufSSyed** [reply]
Oh I see, I was pinching towards the center which didn't do anything, I'm supposed to pinch away from it like the image shows.

By the way, is there a way to change the gestures / what they do in the app? That could be a setting where the default is to scrub by swiping with two fingers, but users like me to change it to something else if they'd like. I'd want to swipe with two fingers to zoom instead.

**Reply by mifi** [reply]
unfortunately it's not possible to configure this in losslesscut now. maybe you can use some app in your operating system that allows you to remap gestures

**Reply by YousufSSyed** [reply]
OK, I'd get back to you when or if I get to that.

**Reply by dreirund** [reply]
> *you can use your mouse wheel or the zoom button in the lower left*

Mouse wheel on the waveform (I have loaded an audio only file) does not do anything. Mouse wheel on the timestamp moves in time, does not zoom. Only the "zoom" selector (but not buttons, it is a drop-down) I find.

**Reply by mifi** [reply]
for now you have to use the mouse wheel on the small waveform (right above the timeline)

**Comment by dreirund** [comment]
On Mon, 09 Dec 2024 06:48:02 -0800, Mikael Finstad
***@***.***> wrote:

> for now you have to use the mouse wheel on the small waveform (right
> above the timeline)
> 

There it only says "Zoom in more to view waveform". When I use the
drop-down to zoom in more, I see the small waveform. When I do
mousewheel there, it scrolls in the timeline, does not zoom.

**Reply by mifi** [reply]
Ah, it was wrong of me to say mouse wheel. ~The zoom action in LosslessCut is the HTML `scroll` event. On MacOS (which I use), this can be achieved by pinching two fingers on the trackpad of my macbook and then moving them apart or closer. on Windows I'm not sure what is the shortcut for that if you don't have a trackpad, but I imagine something like CTRL+mousewheel~

EDIT: Actually LosslessCut uses the `wheel` event and checks for the`ctrlKey` property (or whatever modifier key is configured in losslesscut for "Mouse wheel zoom modifier key"). It seems that MacOS simulates `wheel` event with `ctrlKey: true`.

**Comment by dreirund** [comment]
On Mon, 09 Dec 2024 23:44:38 -0800, Mikael Finstad
***@***.***> wrote:

> Ah, it was wrong of me to say mouse wheel. The zoom action in
> LosslessCut is the HTML `scroll` event. [...] on Windows I'm not sure
> what is the shortcut for that if you don't have a trackpad, but I
> imagine something like CTRL+mousewheel

I am neither on MacOS nor on Windows.

I tried CTRL+mousewheel. Works. But _very_ coarse (1x -> 11x -> 32x ->
72x ...). This should go in fine steps.

**Reply by mifi** [reply]
you can configure it "Timeline trackpad/wheel sensitivity"
